### ISSUE TITLE
Charge Reference

### ISSUE THREAD

**Issue by teghoz** [adebisi-fa/paystack-dotnet#12]
```
var chargeRequest = new CardChargeRequest
                    {
                        Amount = (decimal.Parse(card.amount) * 100).ToString(),
                        Email = card.email,
                        Card = new Card
                        {
                            Cvv = card.cvv,
                            ExpiryMonth = card.expirymonth,
                            ExpiryYear = card.expiryyear,
                            Number = card.cardno
                        },
                        Pin = input.PIN,
                        MetadataObject = chargeCustomData,
                        //CustomFields = customFields
                    };
```

How does one pass reference for the transaction - https://developers.paystack.co/reference#charge

**Comment by adebisi-fa** [comment]
Hello @teghoz,

That's an omission.  The library is currently assuming paystack will auto-generate a reference (as it usually does, if not supplied).

We should be allowed to specify a reference (in situation where we want such to be tracked outside paystack, or something).

Thanks for reporting this.

Would be fixed in a moment (and nuget will be updated, too).

Thanks.

**ADEBISI Foluso A.** (.Net)

**Comment by adebisi-fa** [comment]
Fixed in [v0.7.4](https://github.com/adebisi-fa/paystack-dotnet/releases/tag/v0.7.4).
